Sources/BraveWallet/Crypto/NetworkSelectionRootView.swift

```diff
@@ -12,66 +12,204 @@ struct NetworkSelectionRootView: View {
+                .filter { primaryNetwork in
+                  !selectedNetworks.contains(
+                    where: { $0.chainId == primaryNetwork.chainId && $0.coin == primaryNetwork.coin }
+                  )
+                }
+              unselectedNetworks.forEach(selectNetwork)
+            }
+          }
+        )
         ForEach(allNetworks.primaryNetworks) { network in
           Button(action: { selectNetwork(network) }) {
             NetworkRowView(
               network: network,
               selectedNetworks: selectedNetworks
             )
           }
-          .listRowBackground(Color(.secondaryBraveGroupedBackground))
         }
-      }
-      Section(content: {
+        
+        DividerLine()
+          .padding(.top, 12)
+        
+        headerView(
+          title: Strings.Wallet.networkSelectionSecondaryNetworks,
+          allSelected: allSecondarySelected,
+          selectAllAction: {
+            if allSecondarySelected { // deselect all
+              allNetworks.secondaryNetworks.forEach(selectNetwork)
+            } else { // select all
+              let unselectedNetworks = allNetworks.secondaryNetworks
+                .filter { secondaryNetwork in
+                  !selectedNetworks.contains(
+                    where: { $0.chainId == secondaryNetwork.chainId && $0.coin == secondaryNetwork.coin }
+                  )
+                }
+              unselectedNetworks.forEach(selectNetwork)
+            }
+          }
+        )
         ForEach(allNetworks.secondaryNetworks) { network in
           Button(action: { selectNetwork(network) }) {
             NetworkRowView(
               network: network,
               selectedNetworks: selectedNetworks
             )
           }
-          .listRowBackground(Color(.secondaryBraveGroupedBackground))
         }
-      }, header: {
-        WalletListHeaderView(title: Text(Strings.Wallet.networkSelectionSecondaryNetworks))
-      })
-      if Preferences.Wallet.showTestNetworks.value && !allNetworks.testNetworks.isEmpty {
-        Section(content: {
+        
+        if Preferences.Wallet.showTestNetworks.value && !allNetworks.testNetworks.isEmpty {
+          DividerLine()
+            .padding(.top, 12)
+          
+          headerView(
+            title: Strings.Wallet.networkSelectionTestNetworks,
+            allSelected: allTestnetSelected,
+            selectAllAction: {
+              if allTestnetSelected { // deselect all
+                allNetworks.testNetworks.forEach(selectNetwork)
+              } else { // select all
+                let unselectedNetworks = allNetworks.testNetworks
+                  .filter { testNetwork in
+                    !selectedNetworks.contains(
+                      where: { $0.chainId == testNetwork.chainId && $0.coin == testNetwork.coin }
+                    )
+                  }
+                unselectedNetworks.forEach(selectNetwork)
+              }
+            }
+          )
```

Comment:
feel like this is being shared multiple times. can be refactored to avoid code duplication.

Reply:
Good idea, I've made a `SelectAllHeaderView` for re-use with this view (primary networks, secondary networks, test networks headers) and with `AccountSelectionRootView` in [89405ba](https://github.com/brave/brave-ios/pull/7665/commits/89405ba3ef0c8cb5b55ce9ec8f8f7c07643c9833)👍.